nvflare/private/fed/client/command_agent.py

```diff
@@ -36,51 +38,60 @@ def __init__(self, federated_client, listen_port, client_runner) -> None:
         self.asked_to_stop = False
 
         self.commands = AdminCommands.commands
+        self.logger = logging.getLogger(self.__class__.__name__)
 
     def start(self, fl_ctx: FLContext):
-        # self.thread = threading.Thread(target=listen_command, args=[federated_client, int(listen_port), client_runner])
-        self.thread = threading.Thread(target=listen_command, args=[self, fl_ctx])
+        engine = fl_ctx.get_engine()
+        self.thread = threading.Thread(
+            target=listen_command, args=[self.listen_port, engine, self.execute_command, self.logger]
+        )
         self.thread.start()
 
-        pass
-
     def listen_command(self, fl_ctx):
```

Comment:
this `listen_command` is not being used

Reply:
we use the one from `from ..utils.fed_utils import listen_command`

Reply:
Removed.

Reply:
Removed.